src/App.tsx

```diff
@@ -124,7 +124,11 @@ function LargeScreenLayout({ query, route, map, error, mapOptions, info }: Layou
                         selectedProfile={query.routingProfile}
                         autofocus={false}
                     />
-                    <CustomModelBox enabled={query.customModelEnabled} encodedValues={info.encoded_values} />
+                    <CustomModelBox
+                        enabled={query.customModelEnabled}
+                        encodedValues={info.encoded_values}
+                        initialCustomModelStr={query.initialCustomModelStr}
```

Comment:
Note that the `initialCustomModelStr` prop is really what it says: The **initial** string value for the custom model box. It is only ever used when the component is created initially, see also my comments here: https://github.com/graphhopper/graphhopper-maps/pull/183#discussion_r806256275. Not sure if there is a better way to make this clear.

Reply:
> Couldn't we avoid this parameter somehow? 

To put it short: I don't really think so.

> similar to zoom it does not seem to fit well into QueryStore

Yes, but just because we did not create a separate store for this (and maybe zoom) that might be a better fit it does not say we don't need it.

> E.g. we could parse the custom_model URL parameter before we set the QueryStore.custom_model variable and if this parsing fails set it to null? 

The custom_model variable in QueryStore is not the backend for the editor content (see the comment I linked above).